### ISSUE TITLE
Self hosted Prebid Server returns 400

### ISSUE THREAD

**Issue by monofonik** [prebid/prebid-server#458]
We're testing a self hosted install of Prebid server, and attempting to send bid requests to our own AppNexus seat.

The client side adapter appears to send a different type of POST body to the endpoint based on the endpoint URL, even though no other config params are changed.

POST body to endpoint URL https://prebid.adnxs.com/pbs/v1/auction:

```json
{
    "account_id": "7290",
    "tid": "0f129ab9-61f8-41f7-b9b7-d65f2e0095a4",
    "max_bids": 1,
    "timeout_millis": 1000,
    "cache_markup": 0,
    "url": "http://jono-local.pgxyz.net:8080/index.html?pbjs_debug=true",
    "prebid_version": "1.7.0",
    "ad_units": [{
        "code": "ad-unit-1",
        "sizes": [{
            "w": 320,
            "h": 50
        }],
        "bids": [{
            "bidder": "appnexus",
            "params": {
                "placementId": 12913056
            },
            "bid_id": "1dd8054c5252ae"
        }],
        "transactionId": "41f35caa-93cc-4131-9a09-68e331cb5bc5"
    }],
    "is_debug": true,
    "site": {
        "publisher": {
            "id": "7290"
        },
        "page": "http://jono-local.pgxyz.net:8080/index.html?pbjs_debug=true"
    }
}
```

POST body to endpoint URL https://prebid.adnxs.com/pbs/v1/auction: https://prebid.playground.xyz/openrtb2/auction

```json
{
    "id": "3c460187-31fa-4cde-8c2b-748ea5864692",
    "source": {
        "tid": "3c460187-31fa-4cde-8c2b-748ea5864692"
    },
    "tmax": 1000,
    "imp": [{
        "id": "ad-unit-1",
        "ext": {
            "appnexus": {
                "placementId": 12913056
            }
        },
        "banner": {
            "format": [{}]
        }
    }],
    "test": 1,
    "site": {
        "publisher": {
            "id": "7290"
        },
        "page": "http://jono-local.pgxyz.net:8080/index.html?pbjs_debug=true"
    }
}
```

Here's the config object in use:

```javascript
  pbjs.setConfig({
    s2sConfig: {
      accountId: '7290',
      enabled: true,
      bidders: ['appnexus'],
      timeout: 1000,
      adapter: 'prebidServer',
      endpoint: 'https://prebid.playground.xyz/openrtb2/auction',
    }
  });

  var adUnits = [{
    code: 'ad-unit-1',
    sizes: [
      [320, 50]
    ],
    bids: [
      {
        bidder: 'appnexus',
        params: { 
          placementId: '12913056'
        }
      }
    ]
  }];
```

The POST to https://prebid.playground.xyz/openrtb2/auction subsequently returns a 400 error: _Invalid request format: Request imp[0].banner.format[0] should define *either* {w, h} (for static size requirements) *or* {wmin, wratio, hratio} (for flexible sizes) to be non-zero._

Is this a configuration issue or something else?

Cheers

**Comment by jsnellbaker** [comment]
@monofonik This issue was found internally and we have implemented a fix for it that should be included in the upcoming 1.8 Prebid.js release.

**Comment by monofonik** [comment]
Thanks for the prompt action all!
